packages/assets-controllers/src/CurrencyRateController.ts

```diff
@@ -196,23 +197,27 @@ export class CurrencyRateController extends StaticIntervalPollingController<
           error.message.includes('market does not exist for this coin pair')
         )
       ) {
+        // Don't update state on transient / unexpected errors
+        shouldUpdateState = false;
         throw error;
       }
     } finally {
       try {
```

Comment:
Just out of curiosity, what is the reason for this try - finally block inside of the finally block?

Reply:
to be sure the mutex always gets released

Reply:
Yeah, I was wondering if the [update function](https://github.com/MetaMask/core/blob/b9e983011b6bbec73d8ae0e38921f45ab19ae782/packages/base-controller/src/BaseControllerV2.ts#L188), could indeed throw an error